src/MongoDB.Driver.Core/Core/Connections/ConnectionInitializer.cs

```diff
@@ -64,24 +68,35 @@ public ConnectionDescription InitializeConnection(IConnection connection, Cancel
             return description;
         }
 
-        public async Task<ConnectionDescription> InitializeConnectionAsync(IConnection connection, CancellationToken cancellationToken)
+        public async Task<ConnectionDescription> InitializeConnectionAsync(IConnection connection,
+            CancellationToken cancellationToken)
         {
             Ensure.IsNotNull(connection, nameof(connection));
+            var isMasterCommand = AddAuthenticatorCustomizationsToIsMasterCommand(
+                AddClientDocumentToIsMasterCommand(CreateIsMasterCommand(), _clientDocument),
+                connection.Settings.Authenticators);
+            var isMasterProtocol = CreateIsMasterProtocol(isMasterCommand);
+            var isMasterResult = await GetIsMasterResultAsync(connection, isMasterProtocol, cancellationToken)
+                .ConfigureAwait(false);
 
-            var isMasterProtocol = CreateIsMasterProtocol();
-            var isMasterResult = new IsMasterResult(await isMasterProtocol.ExecuteAsync(connection, cancellationToken).ConfigureAwait(false));
 
             var buildInfoProtocol = CreateBuildInfoProtocol();
-            var buildInfoResult = new BuildInfoResult(await buildInfoProtocol.ExecuteAsync(connection, cancellationToken).ConfigureAwait(false));
+            var buildInfoResult =
+                new BuildInfoResult(await buildInfoProtocol.ExecuteAsync(connection, cancellationToken)
+                    .ConfigureAwait(false));
```

Comment:
Nothing changed here but formatting right? I usually try and keep formatting changes to a minimum to keep diffs smaller.

Reply:
Ah, apologies! 
Either (1) something initially changed, and I backed the change out.
(2) Autoformatting took over when I hit enter and I didn't notice.
I'll revert it.